pkgs/applications/science/logic/potassco/clingcon.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, cmake, bison, re2c }:
+
+stdenv.mkDerivation rec {
+  pname = "clingcon";
+  version = "3.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "potassco";
+    repo = "${pname}";
+    rev = "d36fd87c8937b0ae2b39593aab26f3eb717749d3";
+    fetchSubmodules = true;
+    sha256 = "1q7517h10jfvjdk2czq8d6y57r8kr1j1jj2k2ip2qxkpyfigk4rs";
+   };
+
+  postPatch = ''
+    # deal with clingcon through git submodules recursively importing
+    # an outdated version of libpotassco which uses deprecated <xlocale.h> header in .cpp files
+    find ./ -type f -exec sed -i 's/<xlocale.h>/<locale.h>/g' {} \;
+  '';
+
+  nativeBuildInputs = [ cmake bison re2c ];
+
+  cmakeFlags = [
+    "-DCLINGCON_MANAGE_RPATH=ON"
+    "-DCLINGO_BUILD_WITH_PYTHON=OFF"
+    "-DCLINGO_BUILD_WITH_LUA=OFF"
+  ];
+
+  meta = {
+    inherit version;
```

Comment:
I'm not sure this is needed. Did you read it is required in the documentation? Most derivations don't have this.

Reply:
I guess this is following what I did for `clingo` because the non-`meta` `version` attribute gets handled in a pretty messy way and the auto-updater script is easier to write if the version is provided in a clean place.

Of course this makes less sense when `rev` in the `src` is not version-derived anyway.

Reply:
What auto-updater?

Reply:
Well, I sometimes use update-walker when I remember to update Clingo

Reply:
Well, good luck with that :) I tend to count on @r-ryantm to do the dirty job for me when possible...